swri_roscpp/include/swri_roscpp/topic_service_client.h

```diff
@@ -55,7 +55,7 @@ class TopicServiceClientRaw
   int sequence_;
 
 public:
-  TopicServiceClientRaw() : sequence_(0), timeout_(ros::Duration(4.0))
+  TopicServiceClientRaw() : sequence_(rand()), timeout_(ros::Duration(4.0))
```

Comment:
What's the purpose behind putting a random number in here?

Reply:
It ensures that different clients making the same service request in the same node start with requests with different sequence numbers so there's less possible confusion between them.

Perhaps this random number would do better to be appended to the client name in the request instead?